Add ability to pass non-text variables through the cli `--vars` flag

kkr16 opened the pull request:
### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

cboneti commented:
gcbrun/

cboneti commented on pkg/config/config.go:
```diff
@@ -450,7 +450,11 @@ func (dc *DeploymentConfig) SetCLIVariables(cliVariables []string) error {
-		key, value := arr[0], arr[1]
+		// Convert the variable's string litteral to its equivalent default type.
+		var out interface{}
+		yaml.Unmarshal([]byte(arr[1]), &out)
```
Maybe a silly question, what happens if this fails? should you not check for an error?

nick-stroud commented:
/gcbrun

nick-stroud commented:
/gcbrun

heyealex commented on pkg/config/config_test.go:
```diff
@@ -542,12 +543,18 @@ func (s *MySuite) TestSetCLIVariables(c *C) {
 	cliRegion := "cli_region"
 	cliZone := "cli_zone"
 	cliKeyVal := "key=val"
+	cliKeyBool := "true"
```
I'd like to see the full set of requirements from the design being tested here, at least in a simple form:

> Parse simple variables provided under a --vars flag
> - Boolean: true
> - Integers: 3
> - Floats: 3.14
> - Strings: “foo”
>
> Parse complex variables provided under a --vars flag
> - Array: [1, 2, 3]
> - Map: {a: 1, b: 2}
> - Array with maps: [foo, {a: 1, b: 2}]
> - Map with arrays: {foo: [1, 2, 3], bar: [a, b, c]}

kkr16 replied:
Done.

heyealex reviewed (changes requested):
Thanks for the PR and the great addition to the Toolkit!
I added a comment for covering a couple more test cases, once those have been added this should be good to go.

kkr16 commented:
Thanks for the review @heyealex !

I updated the tests cases based on your recommendations. 
I've also updated [cmd/README.md](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/410cf5d9d60e08f20fcf4f5926a2e536579b0313/cmd/README.md#flags---create) with an exhaustive list of tested examples, showing escape characters for bash.

cboneti commented:
gcbrun/

heyealex reviewed (approved):
Thanks for adding the new tests and updating the documentation! Approved
